src/Marten/Events/Daemon/Coordination/ProjectionCoordinator.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JasperFx.Core;
+using Marten.Events.Daemon.Resiliency;
+using Marten.Storage;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+namespace Marten.Events.Daemon.Coordination;
+
+public class ProjectionCoordinator<T>: ProjectionCoordinator, IProjectionCoordinator<T> where T : IDocumentStore
+{
+    public ProjectionCoordinator(T documentStore, ILogger<ProjectionCoordinator> logger) : base(documentStore, logger)
+    {
+    }
+}
+
+public class ProjectionCoordinator : IProjectionCoordinator
+{
+    private readonly StoreOptions _options;
+    private readonly ILogger<ProjectionCoordinator> _logger;
+
+    private readonly ConcurrentDictionary<IMartenDatabase, IProjectionDaemon> _daemons = new();
```

Comment:
**QUESTION:** Is there any reason not to use `ImHashMap` here?

Reply:
Yes, I purposely used ConcurrentDictionary here. We're not accessing it often enough for the perf to matter, but the big thing was to never end up with an orphaned object instance of an IProjectionDaemon. We could use ImHashMap, but we'd need to add locking around writes to it. Or whatever the built in wrapper around it it. I just thought it was easier to use ConcurrentDictionary here.

Reply:
Scratch what I said, switching to ImHashMap & a lock now.